internal/provider/data_source_archive_file_tgz_test.go

```diff
@@ -0,0 +1,1124 @@
+					),
+					r.TestCheckResourceAttr(
+						"data.archive_file.foo", "output_sha", "d2f26a69cbb920715f797f81c1477c41d8fc9195",
+					),
+					r.TestCheckResourceAttr(
+						"data.archive_file.foo", "output_sha256", "6f1b1a5d17e42fae154f0bdf9301a0ad43394d7fe8485b64dfdb533a0cf07784",
+					),
+					r.TestCheckResourceAttr(
+						"data.archive_file.foo", "output_base64sha256", "bxsaXRfkL64VTwvfkwGgrUM5TX/oSFtk39tTOgzwd4Q=",
+					),
+					r.TestCheckResourceAttr(
+						"data.archive_file.foo", "output_sha512", "e0dc636c3d11095a5b5e97c598185ee3d8a0ed7d1accb69cc28419aeeaeda22b2e774a260f71892a2e85efae1a3aee36669b61dafaed9ac0886d2ca8c5add6e9",
+					),
+					r.TestCheckResourceAttr(
+						"data.archive_file.foo", "output_base64sha512", "4NxjbD0RCVpbXpfFmBhe49ig7X0azLacwoQZrurtoisud0omD3GJKi6F764aOu42Zpth2vrtmsCIbSyoxa3W6Q==",
+					),
+				),
+			},
+			{
+				Config: testAccArchiveFileDirConfig("tar.gz", f),
+				Check: r.ComposeTestCheckFunc(
+					testAccArchiveFileSize(f, &fileSize),
+					r.TestCheckResourceAttrPtr("data.archive_file.foo", "output_size", &fileSize),
+					r.TestCheckResourceAttr(
+						"data.archive_file.foo", "output_md5", "6678fae1fe2077c767bac136861e3bdc",
+					),
+					r.TestCheckResourceAttr(
+						"data.archive_file.foo", "output_sha", "3af6ef3c57aaa5ab3681cd25f916d6651b806cb6",
+					),
+					r.TestCheckResourceAttr(
+						"data.archive_file.foo", "output_sha256", "1b10e0f355025819486fb688aa04217939ea976cd271089bc0092e2994dbaaba",
+					),
+					r.TestCheckResourceAttr(
+						"data.archive_file.foo", "output_base64sha256", "GxDg81UCWBlIb7aIqgQheTnql2zScQibwAkuKZTbqro=",
+					),
+					r.TestCheckResourceAttr(
+						"data.archive_file.foo", "output_sha512", "adb56fca1e40420d4f994d031a08ca0d1ee51783f3c5d1631b6ed2b460ff2577f9154cb5f1c06edd0b0162899f7cfa7cc3d1f02ec9c9ae76f7ea64a31ba8cb81",
+					),
+					r.TestCheckResourceAttr(
+						"data.archive_file.foo", "output_base64sha512", "rbVvyh5AQg1PmU0DGgjKDR7lF4PzxdFjG27StGD/JXf5FUy18cBu3QsBYomffPp8w9HwLsnJrnb36mSjG6jLgQ==",
+					),
+				),
+			},
+			{
+				Config: testAccArchiveFileDirExcludesConfig("tar.gz", f),
+				Check: r.ComposeTestCheckFunc(
+					testAccArchiveFileSize(f, &fileSize),
+					r.TestCheckResourceAttrPtr("data.archive_file.foo", "output_size", &fileSize),
+				),
+			},
+			{
+				Config: testAccArchiveFileDirExcludesGlobConfig("tar.gz", f),
+				Check: r.ComposeTestCheckFunc(
+					testAccArchiveFileSize(f, &fileSize),
+					r.TestCheckResourceAttrPtr("data.archive_file.foo", "output_size", &fileSize),
+				),
+			},
+			{
+				Config: testAccArchiveFileMultiSourceConfig("tar.gz", f),
+				Check: r.ComposeTestCheckFunc(
+					testAccArchiveFileSize(f, &fileSize),
+					r.TestCheckResourceAttrPtr("data.archive_file.foo", "output_size", &fileSize),
+				),
+			},
+		},
+	})
+}
+
+func TestAccTarGzArchiveFile_SourceConfigMissing(t *testing.T) {
+	for _, format := range []string{"tar.gz", "tar.gz"} {
```

Comment:
Can you remove this loop now that the tests are separate? Same comment applies to `TestAccTarGzArchiveFile_SourceConfigConflicting()` and the equivalent test functions in `data_source_archive_file_zip_test.go`.

Reply:
Done